### PR TITLE
[ISSUE-441] Override Drive Health via annotation

### PR DESCRIPTION
**Pull request by safronovD**
### Issue #441

- Override Drive Health if `health` annotation is set
- Generate event `DriveHealthOverridden`
- Generate event `DriveHealth<>` after deleting annotation

## PR checklist
- [x] Add link to the issue
- [x] Choose PR label
- [x] New unit tests added
- [x] Modified code has meaningful comments
- [ ] All TODOs are linked with the issues
- [x] All comments are resolved

## Testing
```
user@user-vm ~/g/s/g/d/csi-baremetal-operator (master)> kubectl annotate drive 11a64144-c873-435f-b8c5-51aa05268593 health=SUSPECT
drive.csi-baremetal.dell.com/11a64144-c873-435f-b8c5-51aa05268593 annotated

user@user-vm ~/g/s/g/d/csi-baremetal-operator (master)> kubectl get events --sort-by='.lastTimestamp' | grep LOOPBACK2955441738
3m27s       Normal    DriveDiscovered            drive/05bfbccd-0526-4167-8c98-090590f6b67d           New drive discovered SN: LOOPBACK2955441738, Node: 77a7e96c-ee8e-4e35-9110-938686735333.Drive Details: SN='LOOPBACK2955441738', Node='77a7e96c-ee8e-4e35-9110-938686735333', Type='HDD', Model='Test Loopback', Size='105906176', Firmware=''
3m27s       Normal    DriveHealthGood            drive/05bfbccd-0526-4167-8c98-090590f6b67d           Drive health is: GOOD, previous state: UNKNOWN.Drive Details: SN='LOOPBACK2955441738', Node='77a7e96c-ee8e-4e35-9110-938686735333', Type='HDD', Model='Test Loopback', Size='105906176', Firmware=''
2m27s       Warning   DriveHealthOverridden      drive/05bfbccd-0526-4167-8c98-090590f6b67d           Drive health is overridden with: SUSPECT, real state: GOOD.Drive Details: SN='LOOPBACK2955441738', Node='77a7e96c-ee8e-4e35-9110-938686735333', Type='HDD', Model='Test Loopback', Size='105906176', Firmware=''
2m27s       Normal    DriveReadyForReplacement   drive/05bfbccd-0526-4167-8c98-090590f6b67d           Drive is ready for replacement, Drive Details: SN='LOOPBACK2955441738', Node='77a7e96c-ee8e-4e35-9110-938686735333', Type='HDD', Model='Test Loopback', Size='105906176', Firmware=''

user@user-vm ~/g/s/g/d/csi-baremetal-operator (master) [1]> kubectl annotate drive 05bfbccd-0526-4167-8c98-090590f6b67d health-
drive.csi-baremetal.dell.com/05bfbccd-0526-4167-8c98-090590f6b67d annotated

user@user-vm ~/g/s/g/d/csi-baremetal-operator (master)> kubectl get events --sort-by='.lastTimestamp' | grep LOOPBACK2955441738
3m27s       Normal    DriveDiscovered            drive/05bfbccd-0526-4167-8c98-090590f6b67d           New drive discovered SN: LOOPBACK2955441738, Node: 77a7e96c-ee8e-4e35-9110-938686735333.Drive Details: SN='LOOPBACK2955441738', Node='77a7e96c-ee8e-4e35-9110-938686735333', Type='HDD', Model='Test Loopback', Size='105906176', Firmware=''
3m27s       Normal    DriveHealthGood            drive/05bfbccd-0526-4167-8c98-090590f6b67d           Drive health is: GOOD, previous state: UNKNOWN.Drive Details: SN='LOOPBACK2955441738', Node='77a7e96c-ee8e-4e35-9110-938686735333', Type='HDD', Model='Test Loopback', Size='105906176', Firmware=''
2m27s       Warning   DriveHealthOverridden      drive/05bfbccd-0526-4167-8c98-090590f6b67d           Drive health is overridden with: SUSPECT, real state: GOOD.Drive Details: SN='LOOPBACK2955441738', Node='77a7e96c-ee8e-4e35-9110-938686735333', Type='HDD', Model='Test Loopback', Size='105906176', Firmware=''
2m27s       Normal    DriveReadyForReplacement   drive/05bfbccd-0526-4167-8c98-090590f6b67d           Drive is ready for replacement, Drive Details: SN='LOOPBACK2955441738', Node='77a7e96c-ee8e-4e35-9110-938686735333', Type='HDD', Model='Test Loopback', Size='105906176', Firmware=''
56s         Normal    DriveHealthGood            drive/05bfbccd-0526-4167-8c98-090590f6b67d           Drive health is: GOOD, previous state: SUSPECT.Drive Details: SN='LOOPBACK2955441738', Node='77a7e96c-ee8e-4e35-9110-938686735333', Type='HDD', Model='Test Loopback', Size='105906176', Firmware=''
```

### CODE REVIEW

**Review by yurkov-anton — APPROVED** [review]

File: pkg/node/volumemgr.go [hunk 1]
```diff
@@ -1136,3 +1156,12 @@ func (m *VolumeManager) isPVCNeedFakeAttach(volumeID string) bool {
 
 	return false
 }
+
+func (m *VolumeManager) overrideDriveHealth(drive *api.Drive, overriddenHealth, driveCRName string) {
+	if (overriddenHealth == apiV1.HealthSuspect) || (overriddenHealth == apiV1.HealthBad) {
```
Comment (mishoyama) [inline pkg/node/volumemgr.go:1161]: What is the purpose of this check? Can't user mark BAD disk as GOOD?

File: pkg/node/volumemgr.go [hunk 2]
```diff
@@ -1136,3 +1156,12 @@ func (m *VolumeManager) isPVCNeedFakeAttach(volumeID string) bool {
 
 	return false
 }
+
+func (m *VolumeManager) overrideDriveHealth(drive *api.Drive, overriddenHealth, driveCRName string) {
```
Comment (mishoyama) [inline pkg/node/volumemgr.go:1160]: Please add comments.

File: pkg/node/volumemgr.go [hunk 3]
```diff
@@ -1031,6 +1042,15 @@ func (m *VolumeManager) createEventForDriveStatusChange(
 		statusMsgTemplate, currentStatus, prevStatus)
 }
 
+func (m *VolumeManager) createEventForDriveHealthOverridden(
```
Comment (mishoyama) [inline pkg/node/volumemgr.go:1045]: Please add comments.

File: pkg/node/volumemgr.go [hunk 4]
```diff
@@ -67,6 +67,8 @@ const (
 
 	fakeAttachAnnotation = "pv.attach.kubernetes.io/ignore-if-inaccessible"
 	fakeAttachAllowKey   = "yes"
+
+	driveHealthOverrideAnnotation = "health"
```
Comment (mishoyama) [inline pkg/node/volumemgr.go:71]: please put comments for this annotations

Comment (mishoyama) [inline pkg/node/volumemgr.go:71]: I think we also need to add this to DR doc.

**Review by mishoyama — APPROVED** [review]

**Review by mishoyama — CHANGES_REQUESTED** [review]
Approved by mistake. Please address review comments.

File: pkg/node/volumemgr.go (hunk 1 above)
Comment (safronovD) [reply pkg/node/volumemgr.go]: Check with HealthGood and HealthUnknown too

File: pkg/node/volumemgr.go (hunk 2 above)
Comment (safronovD) [reply pkg/node/volumemgr.go:1165]: added

File: pkg/node/volumemgr.go (hunk 3 above)
Comment (safronovD) [reply pkg/node/volumemgr.go:1048]: added

File: pkg/node/volumemgr.go (hunk 4 above)
Comment (safronovD) [reply pkg/node/volumemgr.go:73]: added

Comment (safronovD) [reply pkg/node/volumemgr.go:73]: added

**Review by mishoyama — APPROVED** [review]

**Pull request merged**